src/BenchmarkDotNet/Validators/RuntimeValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using BenchmarkDotNet.Jobs;
+
+namespace BenchmarkDotNet.Validators;
+
+/// <summary>
+/// Validator for runtime characteristic.
+/// <see href="https://github.com/dotnet/BenchmarkDotNet/issues/2609" />
+/// </summary>
+public class RuntimeValidator : IValidator
+{
+    public static readonly IValidator DontFailOnError = new RuntimeValidator();
+
+    private RuntimeValidator() { }
+
+    public bool TreatsWarningsAsErrors => false;
+
+    public IEnumerable<ValidationError> Validate(ValidationParameters input)
+    {
+        var allBenchmarks = input.Benchmarks.ToArray();
+
+        var runtimes = allBenchmarks.Select(x => x.Job.Environment.Runtime)
+                                    .Distinct()
+                                    .ToArray();
+
+        if (runtimes.Length > 1 && runtimes.Contains(null))
```

Comment:
I think we should check if any job contains non-null runtime, and any job contains null runtime.